addon/components/inputs/checkbox-array.js

```diff
@@ -0,0 +1,47 @@
+import AbstractInput from './abstract-input'
+import computed, {readOnly} from 'ember-computed-decorators'
+import layout from 'ember-frost-bunsen/templates/components/frost-bunsen-input-checkbox-array'
+
+export default AbstractInput.extend({
+  // == Component Properties ===================================================
+
+  classNames: [
+    'frost-bunsen-input-checkbox-array',
+    'frost-field'
+  ],
+
+  layout,
+
+  // == State Properties =======================================================
+
+  selected: [],
+
+  // == Computed Properties ====================================================
+
+  @readOnly
+  @computed('bunsenModel')
+  options (bunsenModel) {
+    return bunsenModel.items.enum.map((key) => {
+      return {
+        value: key
+      }
+    })
+  },
+
+  // == Functions ==============================================================
+  /**
+   * This should be overriden by inherited inputs to convert the value to the appropriate format
+   * @param {Boolean|String} data - value to parse
+   * @returns {any} parsed value
+   */
+  parseValue (data) {
+    var selected = this.get('selected')
```

Comment:
I am getting the reference to the **selected** state variable so that I didn't have to put this.get('selected') every time. Unless I'm crazy and the array isn't actually passing by reference, but it seemed to be. If you think that's not clear let me know and I'll just put this.get('selected'). Or is there a better way?

Unless you mean the whole block, in which case **data.value** is giving true/false if the checkbox is checked/unchecked respectively. So in the case that the checkbox is checked I push the _value_ of the checkbox (stored in **data.id**, coming from _value=option.value_ in the template) to the **selected** state array otherwise I remove it.

I'm fairly new to JavaScript as well, so if I'm doing something badly let me know.